### DISCUSSION TITLE
Create a custom button and trigger google signin action

### DISCUSSION THREAD

**Discussion by alexjar2211** [abacritt/angularx-social-login#726] [Ideas]
Hi, It seems there is a way to trigger the action for google signin method. 
https://medium.com/@leonardosalles/a-guide-to-custom-google-sign-in-button-e7b02c2c5e4f

I read that article and applied some changes to replicate the use case using this library and It worked.

If you use `<asl-google-signin-button style="display: none;" #customGoogleBtn></asl-google-signin-button>` it shows that div that menctions in the article and the "click" event can be trigger from the component.

![image](https://github.com/abacritt/angularx-social-login/assets/42119983/e19fdb77-0f61-4217-a0c9-a46a4122cf69)

If you delete "display: none" the button renders and that div stays empty.

![image](https://github.com/abacritt/angularx-social-login/assets/42119983/177177a9-017a-42e4-955c-595cdeccb11e)

Maybe this is not a stable solution because there is not a official documentation, maybe is bug and may google it fixed later. 

Anyhow

``` .html
<asl-google-signin-button style="display: none;" #customGoogleBtn></asl-google-signin-button>
<button class="custom-google-btn" (click)="triggerGoogleAction()">Test</button>
```

``` .ts
@ViewChild('customGoogleBtn') customGoogleBtn: ElementRef;

constructor(private authSocial: SocialAuthService) { }

triggerGoogleAction() {
  this.customGoogleBtn.nativeElement.querySelector('div[role="button"]').click();
}
```

**Comment by pedromartins94** [comment]
I tried to make this work but this does fail in Firefox. Any updates on this?

**Reply by alexjar2211** [reply]
You are right. I have only made it work in chrome. I'm going to close this topic because it is not stable.
